servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/AbstractH2DuplexHandler.java

```diff
@@ -75,7 +69,10 @@ public final void userEventTriggered(ChannelHandlerContext ctx, Object evt) {
 
     final void writeMetaData(ChannelHandlerContext ctx, HttpMetaData metaData, Http2Headers h2Headers,
                              ChannelPromise promise) {
-        endStream = !mayHaveTrailers(metaData) && isPayloadEmpty(metaData);
+        final boolean endStream = emptyMessageBody(metaData);
+        if (endStream) {
+            closeHandler.protocolPayloadEndOutbound(ctx, promise);
```

Comment:
`CloseHandler` API doesn't expect this method will be called multiple times. The `emptyMessageBody` method seems like an indirect way to infer there will be no further writes, but feels a bit "fragile" as it is disconnected from the actual write sequencing. I guess if we add another `concat(EMPTY_TRAILERS)` into the mix, we would already have problems bcz we would be writing on a stream we already set "endOfStream" on...

Reply:
Correct, I can keep `endStream` as a class variable as an additional protection from further writes.

Reply:
Discussed offline, additional protection is not necessary because netty will fire an exception anyway if we write after `endStream` was already written. Reverted: https://github.com/apple/servicetalk/pull/1679/commits/6cea6d42a70f475c9c2d9aac00f0ec22ff215509